docs/primers/tables.md

```diff
@@ -9,6 +9,267 @@ Tablespecs
+
+```javascript
+{
+    ...
+    "computed_fields": [
+        {
+            "fieldName": "foo",
+            "value": {
+                "replace": {
+                    "search": "foo",
+                    "replace": "bar",
+                    "subject": "$x"
+                },
+                "arithmetic":["$x", "-", 4"]
+            }
+        }
+    ]
+}
+```
+
+Fields are referenced as variables defined by the fieldName prefixed with ``` $ ```: ``` "fieldName": "foo" ``` would be ``` $foo ```, etc.
+
+Conditional functions
+---------------------
+
+Conditionals let you set the value of the table field based on other values.  Conditionals *must* contain an 'if' property
+and must have either a 'then' or an 'else' property, or both.
+
+"if" expects an array with a length of 1 or 3 as its value: if the array's count is 1, it will just check for the existence
+of the value:
+
+```javascript
+{
+    "fieldName": "foo",
+    "value": {
+        "conditional": {
+            "if": ["$x"] // Does the fieldName 'x' have a value?
+            "then": true,
+            "else": false
+        }
+    }
+}
+```
+
+If the array's count is three, the 2nd value must be one of:
+">","<",">=", "<=", "==", "!=", "contains", "not contains", "~=", "!~"
+there is no strict type comparison.
+
+"then" and "else" values may be of any type, including another function.  To replicate 'else if', you can use another
+ conditional function as the value of 'else'.
+
+ Arithmetic functions
+ --------------------
+ Arithmetic functions allow you to do simple arithmetic with the values.  The function will try to cast the value to the
+ most appropriate type (first it will try float, then integer).
+
+ The arithmetic function's value is an array with 3 values: the first and last values can be arrays to allow for order of operations.
+
+ ```javascript
+ {
+    "fieldName": "foo",
+    "value": {
+        "arithmetic": [["$a", "+", "$b"], "*", ["$x", "+", "$y"]]
+    }
+ }
+```
+
+The left and right values of the equation can also be nested functions:
+
+ ```javascript
+ {
+    "fieldName": "foo",
+    "value": {
+        "arithmetic": [
+            [{
+                "conditional": {
+                    "if":["$a" ">" "$c"],
+                    "then": "$a",
+                    "else": "$c"
+                }
+            }, "+", "$b"], "*", ["$x", "+", "$y"]
+        ]
+    }
+ }
+```
+
+The second value must be one of:
+"+", "-", "*", "/", "%"
+
+Replace functions
+-----------------
+
+This function allows you to do a string replace.  Its syntax is analogous to PHP's ``` str_replace ``` function.
+
+It *must* have 'search', 'replace', and 'subject' properties.  Like ``` str_replace ```, 'search' and 'replace' can be
+   arrays.
+
+If, for this example, ``` $x ``` is "bibo:Article"
+
+```javascript
+{
+    "fieldName": "foo",
+    "value": {
+        "replace": {
+            "search": "bibo:",
+            "replace": "",
+            "subject": "$x"
```

Comment:
I think that "!concatenatedType" is the name of the field - just confused me a bit as it looked like it could be an operator..